python/pynessie/commands/merge.py

```diff
@@ -70,5 +71,12 @@ def merge(ctx: ContextObject, ref: str, force: bool, expected_hash: str, hash_on
             """Either condition or force must be set. Condition should be set to a valid hash for concurrency
             control or force to ignore current state of Nessie Store."""
         )
-    ctx.nessie.merge(from_ref, ref, hash_on_ref, expected_hash)
-    click.echo()
+    merge_response = ctx.nessie.merge(from_ref, ref, hash_on_ref, expected_hash)
+    if ctx.json:
```

Comment:
we should add tests to both code paths i.e. verify the output of the merge command with and without the json flag

also since the VCR recording still returns an empty string:
https://github.com/projectnessie/nessie/blob/ad496c3334cee883f4eed949dd666739c06a6368/python/tests/cassettes/test_nessie_cli/test_merge.yaml#L295-L319

most likely the existing test will still print nothing when calling merge, so we need to adjust this

Reply:
Added test cases for both json and normal output and updated cassettes for these